mslice/models/workspacemanager/mantid_workspace_provider.py

```diff
@@ -29,375 +28,393 @@
-        emax = -np.min(en) if (str(ws.getEMode()) == 'Direct') else np.max(en)
-        qmin, qmax, qstep = self.get_q_limits(theta, emax, efix)
-        self.set_limits(ws_name, qmin, qmax, qstep, theta, np.min(en), np.max(en), np.mean(np.diff(en)))
-
-    def process_limits_event(self, ws, ws_name, efix):
-        e_dim = ws.getDimension(ws.getDimensionIndexByName('DeltaE'))
-        emin  = e_dim.getMinimum()
-        emax = e_dim.getMaximum()
-        theta = self._get_theta_for_limits_event(ws)
-        estep = self._original_step_size(ws)
-        emax_1 = -emin if (str(self.get_EMode(ws)) == 'Direct') else emax
-        qmin, qmax, qstep = self.get_q_limits(theta, emax_1, efix)
-        self.set_limits(ws_name, qmin, qmax, qstep, theta, emin, emax, estep)
-
-    def _original_step_size(self, workspace):
-        rebin_history = self._get_algorithm_history("Rebin", workspace.getHistory())
-        params_history = self._get_property_from_history("Params", rebin_history)
-        return float(params_history.value().split(',')[1])
-
-    def _get_algorithm_history(self, name, workspace_history):
-        histories = workspace_history.getAlgorithmHistories()
-
-        for history in reversed(histories):
-            if history.name() == name:
-                return history
-        return None
 
-    def _get_property_from_history(self, name, history):
-        for property in history.getProperties():
-            if property.name() == name:
-                return property
-        return None
+class Axis(object):
+    def __init__(self, units, start, end, step):
+        self.units = units
+        self.start = start
+        self.end = end
+        self.step = step
+
+    def __eq__(self, other):
+        # This is required for Unit testing
+        return self.units == other.units and self.start == other.start and self.end == other.end \
+            and self.step == other.step and isinstance(other, Axis)
+
+    def __repr__(self):
+        info = (self.units, self.start, self.end, self.step)
+        return "Axis(" + " ,".join(map(repr, info)) + ")"
+
+
+def get_workspace_handle(workspace_name):
+    """"Return handle to workspace given workspace_name_as_string"""
+    # if passed a workspace handle return the handle
+    if isinstance(workspace_name, Workspace):
+        return workspace_name
+    return _loaded_workspaces[workspace_name]
+
+
+def run_alg(alg_name, output_name=None, store=True, **kwargs):
+    if 'InputWorkspace' in kwargs.keys() and isinstance(kwargs['InputWorkspace'], Workspace):
+        kwargs['InputWorkspace'] = kwargs['InputWorkspace'].raw_ws
+    if output_name is not None:
+        kwargs['OutputWorkspace'] = output_name
+
+    ws = getattr(mantid_algs, alg_name)(**kwargs)
+
+    if store:
+        ws = wrap_workspace(ws, output_name)
+    return ws
+
+
+def get_workspace_names():
+    return [key for key in _loaded_workspaces.keys() if key[:2] != '__']
```

Comment:
I would suggest using [`six.iterkeys`](https://pythonhosted.org/six/#six.iterkeys) to avoid making a copy of the `keys()` in Python 2